synth_data_gen.py

```diff
@@ -0,0 +1,89 @@
+import os
+import argparse
+import asyncio
+import google.generativeai as genai
+
+from utils import is_push_to_hf_hub_enabled, update_args
+from src.pipeline.synth_data_gen import (
+    synth_data_generation, collage_as_dataset
+)
+from src.pipeline.utils import push_to_hf_hub
+
+async def synth_data_gen(args):
+    """
+    synth_data_gen generates synthetic dataset that look similar to the original dataset.
+    its main job is to call synth_data_generation() and collage_as_dataset() in order.
+
+    synth_data_generation() generates and save synthetic dataset in external JSON files,
+    then collage_as_dataset() re-organize into Hugging Face Dataset object.
+
+    Additionally it goes through arguments' validation, and 
+    it pushes generated evaluations to the specified Hugging Face Dataset repo.
+    """    
+    if args.gemini_api_key is not None:
+        genai.configure(api_key=args.gemini_api_key)
+
+    hf_hub = is_push_to_hf_hub_enabled(
+        args.push_synth_ds_to_hf_hub,
+        args.synth_ds_id, args.synth_ds_split, args.hf_token
+    )
+    filenames = await synth_data_generation(
+        args.reference_ds_id, args.reference_ds_split, args.num_samples,
+        args.topic, args.prompt_tmpl_path,
+        args.service_model_name, args.gen_workers,
+        args.save_dir_path
+    )
+    dataset = collage_as_dataset(
+        filenames, args.service_model_name, args.topic, args.synth_ds_split
+    )
+
+    if hf_hub is True:
```

Comment:
If it's false then we can simply save the dataset locally by calling `save_to_disk()`.

Reply:
saved dataset via `save_to_disk()` can by loaded up in the same way?

Reply:
Yes. You can verify too.